### DISCUSSION TITLE
How long should insertion take for 10000 records with 1536 dimensions?

### DISCUSSION THREAD

**Discussion by lukezx3** [milvus-io/milvus#34823] [Q&A and General discussion]
I want to insert 500k records with vectors dimensions=1536, but system always throws an error. How can i increase max insertion data? 

Additionally, I then tried to insert 10k records at a time (same dimensions=1536), and each insert seems to take 2-3 minutes which is much slower than I think it's supposed to take. How can I fix/optimize this?

**Comment by ashkrisk** [comment]
@lukezx3 for ingesting large datasets [bulk ingestion](https://milvus.io/api-reference/pymilvus/v2.4.x/ORM/utility/do_bulk_insert.md) is usually _much_ faster than batch ingestion, and you don't have to split the data into small chunks.

If bulk ingestion won't work for you for whatever reason, I've found that I can speed up batch ingestion quite a bit by inserting concurrently with multiple processes at once, so you might find that worth looking into.

**Reply by lukezx3** [reply]
Hey! Ive tried to do that as well but then milvus etcd container gets unhealthy or milvus standalone shuts down all of a sudden. How do I fix those or find root cause? I’m on 12 GB RAM, 4 core Ubuntu 22.04 VMware machine

**Reply by ashkrisk** [reply]
Can you check the logs for Milvus standalone and etcd? Might provide some insight to what's happening.

**Reply by ashkrisk** [reply]
I see you've already created an issue https://github.com/milvus-io/milvus/issues/34821 about the crash. It's not clear to me from the log what the issue is, but let's wait for a reply from some of the folks who've already commented there.

**Comment by xiaofan-luan** [comment]
> I want to insert 500k records with vectors dimensions=1536, but system always throws an error. How can i increase max insertion data?
> 
> Additionally, I then tried to insert 10k records at a time (same dimensions=1536), and each insert seems to take 2-3 minutes which is much slower than I think it's supposed to take. How can I fix/optimize this?

12GB seems to be enough for a standalone to host 500K 1536 dim records.  maybe you should check about your disk and it's capacity.

by default milvus /var/lib and etcd use default.etcd(better change it to your own directory)

**Reply by lukezx3** [reply]
hi i think youve respodned to the issue ticket ive created also but yeah i agree i thought my setup should be able to insert easily and have no etcd or standalone container issues. hopefully you can help me pinpoint root cause. i think it is related to some memory leak or memory related issue because im tracking insertion time and after 50k records insertion, the time increase exponentially so something gets bottlenecked
